README.md

```diff
@@ -407,6 +407,165 @@ after a sequence of diffs are handled.
+```
+
+Fetches the ops for multiple documents in a collection between the requested version numbers, where the `from` value is inclusive, but the `to` value is non-inclusive.
+
+* `agent` _Agent_: connection agent to pass to the middleware
+* `index` _string_: the name of the target collection or projection
+* `id` _string_: the document ID
+* `fromMap` _Object_: an object whose keys are the IDs of the target documents. The values are the first versions requested of each document. For example, `{abc: 3}` will fetch ops for document with ID `abc` from version `3` (inclusive)
+* `toMap` _Object_: an object whose keys are the IDs of the target documents. The values are the last versions requested of each document (non-inclusive). For example, `{abc: 3}` will fetch ops for document with ID `abc` up to version `3` (_not_ inclusive)
+* `options`: _Object (optional)_: options can be passed directly to the database driver's `getOpsBulk` inside the `opsOptions` property: `{opsOptions: {metadata: true}}`
+* `callback`: _Function_: a callback with the signature `function (error: Error, opsMap: Object): void;`, where `opsMap` is an object whose keys are the IDs of the requested documents, and their values are the arrays of requested ops, eg `{abc: []}`
+
+#### `fetch`
+
+```javascript
+backend.fetch(agent: Agent, index: string, id: string, options: Object, callback: Function): void;
+```
+
+Fetch the current snapshot of a document.
+
+* `agent` _Agent_: connection agent to pass to the middleware
+* `index` _string_: the name of the target collection or projection
+* `id` _string_: the document ID
+* `options`: _Object (optional)_: options can be passed directly to the database driver's `fetch` inside the `snapshotOptions` property: `{snapshotOptions: {metadata: true}}`
+* `callback`: _Function_: a callback with the signature `function (error: Error, snapshot: Snapshot): void;`, where `snapshot` is the requested snapshot
+
+#### `fetchBulk`
+
+```javascript
+backend.fetchBulk(agent: Agent, index: string, ids: string[], options: Object, callback: Function): void;
+```
+
+Fetch multiple document snapshots from a collection.
+
+* `agent` _Agent_: connection agent to pass to the middleware
+* `index` _string_: the name of the target collection or projection
+* `ids` _string[]_: array of document IDs
+* `options`: _Object (optional)_: options can be passed directly to the database driver's `fetchBulk` inside the `snapshotOptions` property: `{snapshotOptions: {metadata: true}}`
+* `callback`: _Function_: a callback with the signature `function (error: Error, snapshotMap: Object): void;`, where `snapshotMap` is an object whose keys are the requested IDs, and the values are the requested `Snapshot`s
+
+<!-- TODO: Is it okay to have ignored `subscribe`, `subscribeBulk`, `querySubscribe`? -->
+
+<!-- TODO: Should we ignore `queryFetch`? -->
+
+#### `queryFetch`
```

Comment:
Should we include this? Does it offer any extra functionality over just doing this from a client?

Reply:
Keep it.